### PR TITLE
allow for modifying Alert copy-button behavior

### LINKED ISSUE #522: Ability to format "copy alert" functionality
Currently the "copy alert" button copies all alert data (history, etc). It would be great to have the ability to specify a format so that only specific alert fields will be copied when the button is clicked.

Our use case is people sending questions about an alert in email, slack, etc.

Some sort of config in alertad.conf to control what fields are copied would be ideal.

### PR DESCRIPTION
**Pull request by emillen**
Use nunjucks-templates to modify AlertDetail copy-button behavior

**Description**
Sometimes you want the ability to copy an alert so you can paste into a third party system (slack, email, etc) in human-readable way. This PR allows for setting a nunjuck-template used for formatting the JSON before putting it in the clipboard.

Fixes  #522 

**Changes**

- Install [Nunjucks](https://mozilla.github.io/nunjucks/) (jinja2-like templating engine)
- When `clipboard_template` setting is set, the webui uses its contents as template for the AlertDetail copy button
- If `clipboard_template` is not set, it defaults to previous behavior

If you think this is a good idea and merge this, I will add PR's for adding the setting in the backend, and also add it to the docs.

---
**Ex:**

Provided following alert-json
```json
{
  "text": "this is a major alert",
  "environment": "prod"
}
```
and this template
```json
{
  "clipboard_template": "[{{environment | upper}}] Alert!\n{{text}}"
}
```
would put the following text in your clipboard
```
[PROD] Alert!
this is a major alert
```

### CODE REVIEW

**Review by kennhardy — APPROVED** [review]
Looks good. Approved together with https://github.com/alerta/alerta/pull/1814

File: src/components/AlertDetail.vue [hunk 1]
```diff
@@ -1021,10 +1022,15 @@ export default {
     close() {
       this.$emit('close')
     },
-    clipboardCopy(text) {
+    clipboardCopy(obj) {
```
Comment (satterly) [inline src/components/AlertDetail.vue:1025]: Can you call the parameter `item` instead of `obj`?

**Review by satterly — CHANGES_REQUESTED** [review]

**Comment by emillen** [comment]
What do you say - does this need more work, or are we fine like this?

**Comment by satterly** [comment]
Actually ignore that. There's probably nothing different. I'll merge.

**Pull request merged**